pkg/aws/core_test.go

```diff
@@ -302,10 +306,26 @@ var _ = Describe("MachineServer", func() {
 					errMessage:        "machine codes error: code = [Internal] message = [Region doesn't exist while trying to create session]",
 				},
 			}),
+			Entry("Placement object with affinity, tenancy and availablityZone set", &data{
+				action: action{
+					machineRequest: &driver.CreateMachineRequest{
+						Machine:      newMachine(0, annotations),
+						MachineClass: newMachineClass(providerSpec),
+						Secret:       providerSecret,
+					},
+				},
+				expect: expect{
+					errToHaveOccurred: false,
+					machineResponse: &driver.CreateMachineResponse{
+						ProviderID: "aws:///eu-west-1/i-0123456789-0",
```

Comment:
```suggestion
						ProviderID: "aws:///eu-west-1/i-0123456789-0/placement={affinity:host,availabilityZone:eu-west-1a,tenancy:host}"
```
You can update the mocking function to return something like this for validating this test case.